packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -522,219 +594,566 @@ class Navigator extends StatefulWidget {
+  /// [NavigatorObserver.didPush]).
+  ///
+  /// Ongoing gestures within the current route are canceled when a new route is
+  /// pushed.
+  ///
   /// Returns a [Future] that completes to the `result` value passed to [pop]
   /// when the pushed route is popped off the navigator.
   ///
+  /// The `T` type argument is the type of the return value of the route.
+  /// {@endtemplate}
+  ///
+  /// ## Sample code
+  ///
   /// Typical usage is as follows:
   ///
   /// ```dart
   /// Navigator.pushNamed(context, '/nyc/1776');
   /// ```
-  static Future<dynamic> pushNamed(BuildContext context, String routeName) {
-    return Navigator.of(context).pushNamed(routeName);
+  @optionalTypeArgs
+  static Future<T> pushNamed<T extends Object>(BuildContext context, String routeName) {
+    return Navigator.of(context).pushNamed<T>(routeName);
   }
 
-  /// Adds the given route to the history of the navigator that most tightly
-  /// encloses the given context, and transitions to it.
+  /// Replace the current route of the navigator that most tightly encloses the
+  /// given context by pushing the route named [routeName] and then disposing
+  /// the previous route once the new route has finished animating in.
   ///
-  /// The new route and the previous route (if any) are notified (see
+  /// {@template flutter.widgets.navigator.pushReplacementNamed}
+  /// If non-null, `result` will be used as the result of the route that is
+  /// removed; the future that had been returned from pushing that old route
+  /// will complete with `result`. Routes such as dialogs or popup menus
+  /// typically use this mechanism to return the value selected by the user to
+  /// the widget that created their route. The type of `result`, if provided,
+  /// must match the type argument of the class of the old route (`TO`).
+  ///
+  /// The route name will be passed to the navigator's [onGenerateRoute]
+  /// callback. The returned route will be pushed into the navigator.
+  ///
+  /// The new route and the route below the removed route are notified (see
   /// [Route.didPush] and [Route.didChangeNext]). If the [Navigator] has any
   /// [Navigator.observers], they will be notified as well (see
-  /// [NavigatorObserver.didPush]).
+  /// [NavigatorObserver.didReplace]). The removed route is notified once the
+  /// new route has finished animating (see [Route.didComplete]).
   ///
   /// Ongoing gestures within the current route are canceled when a new route is
   /// pushed.
   ///
   /// Returns a [Future] that completes to the `result` value passed to [pop]
   /// when the pushed route is popped off the navigator.
+  ///
+  /// The `T` type argument is the type of the return value of the new route,
+  /// and `TO` is the type of the return value of the old route.
+  /// {@endtemplate}
+  ///
+  /// ## Sample code
+  ///
+  /// Typical usage is as follows:
+  ///
+  /// ```dart
+  /// Navigator.pushReplacementNamed(context, '/jouett/1781');
+  /// ```
```

Comment:
would describe somewhere whether the exit animation on the replaced route is run

Reply:
Added: `The removed route's exit animation is not run (see [popAndPushNamed] for a variant that does animated the removed route).`